crates/provider/src/fillers/gas.rs

```diff
@@ -196,8 +197,11 @@ where
     type Fillable = u128;
 
     fn status(&self, tx: &<N as Network>::TransactionRequest) -> FillerControlFlow {
-        // nothing to fill if non-eip4844 tx or max_fee_per_blob_gas is already set
-        if tx.blob_sidecar().is_none() || tx.max_fee_per_blob_gas().is_some() {
+        // Nothing to fill if non-eip4844 tx or `max_fee_per_blob_gas` is already set to a valid
+        // value.
+        if tx.blob_sidecar().is_none()
+            || tx.max_fee_per_blob_gas().is_some_and(|gas| gas >= BLOB_TX_MIN_BLOB_GASPRICE)
```

Comment:
If the field is set to `Some(0)` (which happens because of serialization) we force setting it to `.next_block_blob_fee()` as 0 is always invalid as it is less than `BLOB_TX_MIN_BLOB_GASPRICE` (1).